interface/operations/operations_client.go

```diff
@@ -533,10 +541,10 @@ func (c Client) operation(op requests.OpCode, provider requests.ProviderID, requ
 		return err
 	}
 
-	respBody, err := requests.NewResponse(rcvBuf, response)
+	respBody, err := requests.NewResponse(op, rcvBuf, response)
```

Comment:
the request and response are autgenerated protobuf code.  could do it by putting source files into the packages with the auto generated files, but would make clean prior to regeneration more of a fuss.  doable - will create an issue.

Reply:
i'm also not convinced it would reduce chance of mistakes - the pairing only happens in one place at the moment and we're not using the protocol buffer types elsewhere - we'd just move the 'get the wrong opcode' problem from a single file to a whole bunch of files

Reply:
I am not so good in Go, how does it work? The `request` and `response` fields seem to be of type `proto.Message` but you pass each specific generated `Operation` structure. Is it a cast?

I think it's better to leave it this way rather than modifying the generated code! It's not so much effort

Reply:
its how the go protobuf stuff works - basically passing an interface around - all the generated protobuf types implement the interface - it then uses reflection to do the (un)marshalling.

So the types I pass in are generated types implementing the interface spec.